src/analysis/processing/qgsalgorithmb3dmtogltf.cpp

```diff
@@ -98,56 +98,70 @@ QVariantMap QgsB3DMToGltfAlgorithm::processAlgorithm( const QVariantMap &paramet
   tinygltf::Model model;
   QString errors;
   QString warnings;
-  if ( !QgsGltfUtils::loadGltfModel( b3dmContent.gltf, model, &errors, &warnings ) )
+  const bool res = QgsGltfUtils::loadGltfModel( b3dmContent.gltf, model, &errors, &warnings );
+  if ( !errors.isEmpty() )
   {
-    throw QgsProcessingException( QObject::tr( "Error loading B3DM model: %1" ).arg( errors ) );
+    feedback->reportError( errors );
   }
   if ( !warnings.isEmpty() )
   {
     feedback->pushWarning( warnings );
   }
-  feedback->pushDebugInfo( QObject::tr( "Found %1 scenes" ).arg( model.scenes.size() ) );
-
-  const tinygltf::Scene &scene = model.scenes[model.defaultScene];
-  feedback->pushDebugInfo( QObject::tr( "Found %1 nodes in default scene [%2]" ).arg( scene.nodes.size() ).arg( model.defaultScene ) );
-  if ( !scene.nodes.empty() )
+  if ( !res )
   {
-    const int nodeIndex = scene.nodes[0];
-    const tinygltf::Node &gltfNode = model.nodes[nodeIndex];
-    const tinygltf::Mesh &mesh = model.meshes[gltfNode.mesh];
-    feedback->pushDebugInfo( QObject::tr( "Found %1 primitives in default scene node [%2]" ).arg( mesh.primitives.size() ).arg( nodeIndex ) );
-  }
-
-  if ( !b3dmContent.rtcCenter.isNull() )
-  {
-    // transfer B3DM RTC center to GLTF CESIUM_RTC extension
-    tinygltf::Value::Object cesiumRtc;
-    cesiumRtc["center"] = tinygltf::Value( tinygltf::Value::Array
+    // if we can't read the GLTF, then just write the original B3DM content to the output file
```

Comment:
I'd reword this a bit for clarity. I first misunderstood it as meaning writing the original B3DM file as output file (vs. copying the B3DM's GLTF content even though it failed the load check), which perplexed me :)